mempool/src/interface/mock.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(clippy::unwrap_used)]
+
+use crate::error::Error;
+use crate::tx_accumulator::TransactionAccumulator;
+use crate::MempoolEvent;
+use common::chain::{SignedTransaction, Transaction};
+use common::primitives::Id;
+use std::sync::{Arc, Mutex};
+use subsystem::{CallRequest, ShutdownRequest};
+
+use super::mempool_interface::{MempoolInterface, MempoolSubsystemInterface};
+
+#[derive(Clone)]
+pub struct MempoolInterfaceMock {
+    pub add_transaction_called: Arc<Mutex<bool>>,
```

Comment:
I'm not sure if it is really needed, but if in the future we want to have more complex tests, then having counters instead of boolean flags will be more flexible.

Reply:
It wasn't needed for this change so I just went with `bool`, but yes if we need it sometime in the future, it's an easy change.